django/db/models/query.py

```diff
@@ -460,7 +462,39 @@ def _prepare_for_bulk_create(self, objs):
                 obj.pk = obj._meta.pk.get_pk_value_on_save(obj)
             obj._prepare_related_fields_for_save(operation_name='bulk_create')
 
-    def bulk_create(self, objs, batch_size=None, ignore_conflicts=False):
+    def _select_on_conflicts(
+        self, ignore_conflicts, update_conflicts, update_fields, unique_fields
+    ):
+        if ignore_conflicts and update_conflicts:
+            raise IntegrityError(
+                'You can only assign one conflicts plan, ignore_conflicts or update_conflicts'
+            )
+        if ignore_conflicts:
+            self._check_on_conflicts_supported(ON_CONFLICTS_IGNORE, unique_fields)
+            result = ON_CONFLICTS_IGNORE
+        elif update_conflicts:
+            self._check_on_conflicts_supported(ON_CONFLICTS_UPDATE, unique_fields)
+            if not update_fields:
+                raise IntegrityError(
+                    'You need to specify which fields you want to update'
+                )
+            for f in update_fields:
+                try:
+                    self.model._meta.get_field(f)
+                except exceptions.FieldDoesNotExist:
+                    raise IntegrityError(
+                        'Field you specify is not in table: %s' % f
+                    )
+            result = ON_CONFLICTS_UPDATE
+        else:
+            result = ON_CONFLICTS_NONE
+
+        return result
+
+    def bulk_create(
+        self, objs, batch_size=None, ignore_conflicts=False,
+        update_conflicts=False, update_fields=[], unique_fields=[]
```

Comment:
Please make `update_fields` and `unique_fields` have `None` instead of using mutable defaults and do the following inside the function:

```python
        update_fields = update_fields or []
        unique_fields = unique_fields or []